capella_git_hooks/fix_links.py

```diff
@@ -43,6 +56,46 @@ def get_unmodified_tracked_files() -> list[str]:
         return []
 
 
+def search_and_replace(
+    pattern: re.Pattern, value: str, replacement: str
+) -> str | None:
+    """Return the patched value or the unpatched value."""
+    while match := pattern.search(value):
+        index_match = match.span(1)
+        value = value[: index_match[0]] + replacement + value[index_match[1] :]
+    return value
+
+
+def fix_semantic_resources(
+    root: ET._Element, root_repl: str, fragment_repl: str
+) -> bool:
+    """Patch the semantic resource file paths and return a bool."""
+    changed = False
+    danalysis = next(root.iterchildren(DIAGRAM_OVERVIEW_TAG))
```

Comment:
Having a StopIteration escape from a function that's not an iterator is kind of bad practice; nobody really expects that, and it can have effects on the Python runtime (for loops, specifically). Better catch it as early as possible (i.e. right at the `next()` call) and handle it properly here.

```suggestion
    try:
        danalysis = next(root.iterchildren(DIAGRAM_OVERVIEW_TAG))
    except StopIteration:
        return False
```